### PR TITLE
[keyboard] add smk82_pro keyboard.

### PR DESCRIPTION
**Pull request by jonylee1986**
add smk82_pro keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/steed/smk82_pro/config.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#define BT_MODE_ENABLE
+
+#define ENCODER_DEFAULT_POS 0x3
+
+
+/* I2C Config for LED Driver */
+#define DRIVER_COUNT 2
+#define DRIVER_ADDR_1 0b1110100
+#define DRIVER_ADDR_2 0b1110111
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/config.h:19]:
```suggestion
/* I2C Config for LED Driver */
#define SNLED27351_I2C_ADDRESS_1 SNLED27351_I2C_ADDRESS_GND
#define SNLED27351_I2C_ADDRESS_2 SNLED27351_I2C_ADDRESS_VDDIO
```

File: keyboards/steed/smk82_pro/config.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
+#define BT_MODE_ENABLE
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/config.h:11]:
```suggestion
```

File: keyboards/steed/smk82_pro/info.json [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        "command": false,
+        "console": false,
+        "dip_switch": true,
+        "encoder": true,
+        "extrakey": true,
+        "haptic": false,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/info.json:33]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "dip_switch": true,
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true
```

File: keyboards/steed/smk82_pro/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// clang-format on
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+
+        default:
+            break;
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/keymaps/default/keymap.c:58]:
```suggestion
```

File: keyboards/steed/smk82_pro/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/keymaps/default/keymap.c:61]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/steed/smk82_pro/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    switch (keycode) {
+
+        default:
+            break;
+    }
+    return true;
+}
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/keymaps/via/keymap.c:61]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/steed/smk82_pro/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+// clang-format on
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+
+        default:
+            break;
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/keymaps/via/keymap.c:58]:
```suggestion
```

File: keyboards/steed/smk82_pro/smk82_pro.c [hunk 8]
```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 JoyLee (@itarze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "common/bt_task.h"
+#include <stdlib.h>
```
Comment (zvecr) [inline keyboards/steed/smk82_pro/smk82_pro.c:6]:
```suggestion
#include "rgb_matrix.h"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/steed/smk82_pro/config.h [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+#define ENCODER_DEFAULT_POS 0x3
+
+/* I2C Config for LED Driver */
+#define SNLED27351_I2C_ADDRESS_1 SNLED27351_I2C_ADDRESS_GND
+#define SNLED27351_I2C_ADDRESS_2 SNLED27351_I2C_ADDRESS_VDDIO
+#define I2C1_SDA_PIN B7
+#define I2C1_SCL_PIN B6
+#define I2C1_SCL_PAL_MODE 4
+#define I2C1_OPMODE OPMODE_I2C
```
Comment (drashna) [inline keyboards/steed/smk82_pro/config.h:19]: These are the default values, and don't need to be specified.
```suggestion
```

File: keyboards/steed/smk82_pro/config.h [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+#define I2C1_CLOCK_SPEED 400000
+
+/* SPI Config for spi flash*/
+#define SPI_DRIVER SPIDQ
+#define SPI_SCK_PIN B3
+#define SPI_MOSI_PIN B5
+#define SPI_MISO_PIN B4
+#define SPI_MOSI_PAL_MODE 5
```
Comment (drashna) [inline keyboards/steed/smk82_pro/config.h:27]:
```suggestion
```

File: keyboards/steed/smk82_pro/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/steed/smk82_pro/keymaps/default/keymap.c:1]: Please replace the tabs with spaces.

File: keyboards/steed/smk82_pro/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/steed/smk82_pro/keymaps/via/keymap.c:1]: same

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/steed/smk82_pro/readme.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
+# STEED SMK82 PRO
+![STEED SMK82 PRO](https://github.com/jonylee1986/qmk_firmware_master/assets/108282809/42973b4e-8ed5-43d7-8379-70851c67df0f)
```
Comment (drashna) [inline keyboards/steed/smk82_pro/readme.md:2]: Please rehost on imgur or the like.

File: keyboards/steed/smk82_pro/readme.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
+# STEED SMK82 PRO
+![STEED SMK82 PRO](https://cdn.shopifycdn.net/s/files/1/0809/0550/3006/files/SMK82WARCTIC.jpg?v=1701773689&width=1000)
```
Comment (drashna) [inline keyboards/steed/smk82_pro/readme.md:2]: Please rehost the image.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/steed/smk82_pro/readme.md (hunk 12 above)
Comment (jonylee1986) [reply keyboards/steed/smk82_pro/readme.md]: OK

File: keyboards/steed/smk82_pro/smk82_pro.c [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+// clang-format on
+bool dip_switch_update_kb(uint8_t index, bool active) {
+    if (!dip_switch_update_user(index, active)) {
+        return false;
+    }
+    if (index == 0) {
+        default_layer_set(1UL << (active ? 2 : 0));
+        if (active) {
+            keymap_config.no_gui = 0;
+        }
```
Comment (drashna) [inline keyboards/steed/smk82_pro/smk82_pro.c:131]: wouldn't this work: 
```suggestion
        keymap_config.no_gui = !active;
```

**Pull request closed without merging**